pkg/controller/certificaterequest/certificaterequest_controller.go

```diff
@@ -146,17 +146,12 @@ func (r *ReconcileCertificateRequest) Reconcile(request reconcile.Request) (reco
 	clusterDeploymentName := strings.Split(request.Name, requestNameSplit)[0]            // remove it from the certificate request name
 
 	// fetch the clusterdeployment and bail out if there's an outgoing migration annotation
-	cd := &hivev1.ClusterDeployment{}
-	err = r.client.Get(context.TODO(), types.NamespacedName{Namespace: request.Namespace, Name: clusterDeploymentName}, cd)
+	relocating, err := relocationBailOut(r.client, types.NamespacedName{Namespace: request.Namespace, Name: clusterDeploymentName})
 	if err != nil {
 		return reconcile.Result{}, err
 	}
-
-	// bail out of the loop if there's an outgoing relocation annotation
-	for a, v := range cd.Annotations {
-		if a == hiveRelocationAnnotation && strings.Split(v, "/")[1] == hiveRelocationOutgoingValue {
-			return reconcile.Result{}, nil
-		}
+	if relocating {
+		return reconcile.Result{}, nil
```

Comment:
Can we add a log message or update the status on the CR?

Reply:
done